File: nomad-samples/python/asset_uploads/exceptions/api_exception_handler.py
import json
import sys

'''
 * API Exception Handler
 *
 * @param {Stream RESPONSE
 * @param {string message
'''
def api_exception_handler(RESPONSE, MESSAGE):
    # Set a async default error message
    ERROR = "Unknown error occurred"

    # Check if we have a response object and error message
    if (RESPONSE == None):
        # If not response then throw async default error or message
        if (MESSAGE == None or len(MESSAGE.strip()) == 0):
            sys.exit(ERROR)
        else:
            sys.exit(MESSAGE)



    # Get the response content type to determine how to parse it
    CONTENT_TYPE = RESPONSE.headers.get("content-type")

    # If response BODY is JSON
    if (CONTENT_TYPE != None and CONTENT_TYPE == "application/json"):
        if RESPONSE.text != None:
            # Read response as JSON
            error = json.loads(RESPONSE.text)
            # If error JSON object has error messages then throw the first
            if (error != None):
                sys.exit(MESSAGE + ": " +  error["message"])

        else:
            # Throw message and response status
            sys.exit(MESSAGE + ": " +  str(RESPONSE.status_code))
    else:
        if RESPONSE.text != "":
            # Response BODY is text
            error = RESPONSE.text

            # Throw error if valid
            if (error and len(error.strip()) > 0):
                sys.exit(MESSAGE + ": " +  error)

        else:
            # Throw message and response status
            sys.exit(MESSAGE + ": " +  str(RESPONSE.status_code))


        # Throw message and response status
        sys.exit(MESSAGE + ": " +  str(RESPONSE.status_code))

File: nomad-samples/common/js/account-authenticaton/exceptions/api-exception-handler.js
/**
 * API Exception Handler
 *
 * @param {Stream} response
 * @param {string} message
 */
export default async function apiExceptionHandler(response, message) {
    // Set a default error message
    let error = "Unknown error occurred";

    // Get interval ID
    const intervalId = sessionStorage.getItem("intervalId");

    // Clear interval
    if (intervalId) {
        clearInterval(intervalId);
    }

    // Check if we have a response object and error message
    if (!response) {
        // If not response then throw default error or message
        if (!message || message.trim().length === 0) {
            throw new Error(error);
        } else {
            throw new Error(message);
        }
    }

    // Get the response content type to determine how to parse it
    const contentType = response.headers.get("content-type");

    // If response body is JSON
    if (contentType && contentType.indexOf("application/json") !== -1) {
        try {
            // Read response as JSON
            error = await response.json();
            // If error JSON object has error messages then throw the first
            if (error && error.errors && error.errors.length > 0) {
                throw new Error(`${message}: ${error.errors[0].message}`);
            }
        } catch (ignore) {
            // Throw message and response status
            throw new Error(`${message}: ${response.status}`);
        }

        // Throw message and response status
        throw new Error(`${message}: ${response.status}`);
    } else {
        try {
            // Response body is text
            error = await response.text();

            // Throw error if valid
            if (error && error.trim().length > 0) {
                throw new Error(`${message}: ${error}`);
            }
        } catch (ignore) {
            // Throw message and response status
            throw new Error(`${message}: ${response.status}`);
        }

        // Throw message and response status
        throw new Error(`${message}: ${response.status}`);
    }
}
